### DISCUSSION TITLE
Programmatic acces to Subversion Properties from Eclipse

### DISCUSSION THREAD

**Discussion by maxweissboeck** [subclipse/subclipse#194] [Q&A]
I'm not sure if this is the right place to ask, so forgive me if I'm wrong.

We use Eclipse to generate/transform content out of Eclipse workspace Files (we use Xtext, but that should not matter).
Now we have the need to also include the Subversion Url: - as seen in the attachment - of the files, so we can directly access the log history from subversion in the generated files.

Any hints on how I can get access to this information from inside Eclipse Runtime if im having the files workspace resource?

Thanks a 
![Screenshot 2021-09-29 17 10 07](https://user-images.githubusercontent.com/35797287/135297797-9fb63dd2-990c-473e-8922-2652c0458e00.png)
lot!

**Comment by markphip** [comment]
Everything you need should be available from the Subclipse core plugin.

To just get the URL, I would start with this static method that takes an `IPath`:

```
     LocalResourceStatus status = SVNWorkspaceRoot.peekResourceStatusFor(path);
     String url = status.getUrlString();
```

This same class has other static methods that give back the various SVNResource classes that we use, but if your goal is to access the log history you can probably skip this step. Given a local path in a working copy, the SVN API or CLI will know how to access the right URL to get the history. If you want to use our API you just want something like this:

```
    SVNRevision start = SVNRevison.HEAD;
    SVNRevision end = new SVNRevision.Number(0); // this might not be correct
    boolean stopOnCopy = true;
    boolean includePaths = false;
    long limit = 100;
    ISVNClientAdapter client = SVNProviderPlugin.getPlugin().getSVNClient();
    ISVNLogMessage[] history = client.getLogMessages(path, start, end, stopOnCopy, includePaths, limit);
```

There are a lot of variants of the getLogMessages method depending on what you want to retrieve. I do not 100% recall how to manage the SVNRevision class that it takes as a required input. If you were using the CLI you likely would provide a range of HEAD:0 to retrieve all of the history for a path. Always use the higher revision first as history should flow backwards from current to beginning not the other way around.

Mark
